### PR TITLE
Fixes #32225 - store akey original environment on the right scope

### PR DESCRIPTION
**Pull request by jturel**
Check the BZ for the basic repro steps. For a slightly more clever test:

- change the env + cv of the activation key
- change to a different tab
- change back to Details tab
- everything should look good (cv+env are accurate, no info messages shown for the CV field)

### CODE REVIEW

**Comment by jturel** [comment]
[test katello]

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/activation-key-details.controller.js [hunk 1]
```diff
@@ -51,9 +51,13 @@ angular.module('Bastion.activation-keys').controller('ActivationKeyDetailsContro
             ];
         };
 
+        $scope.setOriginalEnvironment = function(newEnvironment) {
+            $scope.originalEnvironment = newEnvironment;
+        };
+
         $scope.activationKey = ActivationKey.get({id: $scope.$stateParams.activationKeyId}, function (activationKey) {
-            $scope.$broadcast('activationKey.loaded', activationKey);
```
Comment (chris1984) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/activation-key-details.controller.js:55]: What did the `$scope.$broadcast do`?

**Review by chris1984 — COMMENTED** [review]
Starting to test. I would say if we were not changing activation keys down the road, we should prob add a test, but I am fine without one.

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/activation-key-details.controller.js (hunk 1 above)
Comment (jturel) [reply engines/bastion_katello/app/assets/javascripts/bastion_katello/activation-keys/details/activation-key-details.controller.js:55]: It was responsible for sending an event that another scope could act upon. See above controller => `$scope.$on('activationKey.loaded', function () {` no longer needed!

**Comment by jturel** [comment]
> Starting to test. I would say if we were not changing activation keys down the road, we should prob add a test, but I am fine without one.

no harm in adding some tests. was easier than I thought here- done!

**Review by chris1984 — APPROVED** [review]
ACK, code looks good and it works:

Before patch:

https://nimb.ws/XYNpfC

After patch:

https://nimb.ws/RvbJxi

Thanks for explanining the function question I had.

**Comment by chris1984** [comment]
[test katello]

**Pull request merged**